src/skvhttpclient/test/ClientIntegrationTest.cpp

```diff
@@ -41,6 +41,62 @@ Client client;
 const std::string collectionName = "k23si_test_collection";
 const std::string schemaName = "test_schema";
 
+// Helper functions
+// Verify that two records are equal, the parameters are not const as deserialize is not const function
+void verifyEqual(dto::SKVRecord& first, dto::SKVRecord& second) {
+    first.visitRemainingFields([&second] (const auto& field, auto val1) {
```

Comment:
you should seek(0) on both records here, and also seek(0) when you're done to reset their cursors